applications/xcm-tools.md

```diff
@@ -0,0 +1,162 @@
+# XCM Tools
+
+* **Team Name:** [gmajor](https://github.com/gmajor-encrypt)
+* **Payment Address:** 0xC3094f0ddce699a1Ad9Ef2621DF68Cd297a4c44F(USDC)
+* **[Level](https://github.com/w3f/Grants-Program/tree/master#level_slider-levels):** 2
+
+## Project Overview :page_facing_up:
+
+RFPs [https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/xcm-tool.md](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/xcm-tool.md)
+
+### Overview
+
+XCM (Cross-Consensus Message) is a cross-chain message between parachain and relaychain because the xcm format (
+Instruction, MultiAssets, and MultiLocation) are too complex, and no good tool exists. It has yet to be widely used.
+
+XCM Tools is a tool and an SDK. This library is written in golang. It provides the following functions: sending xcm
+messages, parsing xcm message instructions, getting the execution result of the execution after sending xcm
+
+### Project Details
+
+We only have two libs to choose from, and some functions still need to be included.
+
+[Go-Substrate-RPC-Client](https://github.com/centrifuge/go-substrate-rpc-rient)
+Centrifuge's open source and supports transaction and chain status queries. Scale Codec is a static type. Types analysis
+many
+require users definition, and lack of support for Metadata V14 Types
+[Substrate-API-rpc](https://github.com/itering/substrate-api-rpc), [scale.go](https://github.com/scale.go))
+Subscan open source, support chain status query, Scale Codec supports dynamic types, good serialization, and back-order
+functions. But lack the function of transaction sending.
+After comparison, I chose [Substrate-API-rpc](https://github.com/tering/substrate-api-RPC) as dependence, but because of
+the lack of the function of sending transactions, I also need to change dependence.
+
+1. Improve [substrate-api-rpc](https://github.com/itering/substrate-api-rpc), add extrinsic encode, transaction
```

Comment:
Are you planning to submit these changes to the base repo, or to start and maintain your own fork?

Reply:
I plan to initiate a pull request and merge it into the base repo. Some past [issues](https://github.com/itering/substrate-api-rpc/issues/25) show that many developers need to send transactions feature.

Reply:
That's great to hear. I recommend contacting the owners beforehand. They might have requirements or other plans.